### DISCUSSION TITLE
Some detonate events cannot be resolved to an entity

### DISCUSSION THREAD

**Discussion by Brammz** [saul/demofile#228] [Questions] [answered]
I am trying to get the entity on a detonate event using the `entityid`. For the `hegrenade_detonate` and `smokegrenade_detonate` events, this resolves to an entity. However, for `flashbang_detonate` and `decoy_detonate` this cannot be resolved to an entity. (This is also the case for `molotov_detonate`, but this does not seem to have an `entityid` property, according to https://wiki.alliedmods.net/Counter-Strike:_Global_Offensive_Events#molotov_detonate)

I am running the following code, which results in grenade being undefined for the events mentioned above.
```javascript
demoFile.gameEvents.on('***_detonate', e => {
	let thrower = demoFile.entities.getByUserId(e.userid);
	let grenade = demoFile.entities.entities[e.entityid];
	if (thrower && grenade) {
		let throwerPosition = thrower.position;
		let grenadePostiion = grenade.position;
		console.log('%s threw a grenade from position [%s,%s,%s] to position [%s,%s,%s]', thrower.name, throwerPosition.x, throwerPosition.y, throwerPosition.z,  grenadePostiion.x, grenadePostiion.y, grenadePostiion.z);
			}
    });
```

**Comment by saul** [comment, marked as answer]
I think what is most likely happening is that the grenade is being immediately removed from the world on detonation, instead of at the next tick. I could fix this in the library by moving any entity deletions until after the events have been fired. 

In the meantime, you’re best off listening for grenade projectile ‘remove’ events (on demofile.entities). Look for, CDecoyProjectile, for example, in the netprops.txt file in this repository.

**Comment by RaphaelHippe** [comment]
Hey, 
for me `let grenade = demoFile.entities.entities[e.entityid];` is `undefined`. 
Do you have any idea on why that might be the case? I copied the given example and replaced `'***_detonate'` with `'flashbang_detonate'`

Cheers
Raphael

**Comment by saul** [comment]
@RaphaelHippe have you updated to the latest version of the library?

**Comment by RaphaelHippe** [comment]
Yea that was it. Sorry my bad.
